File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        left,right = 0, len(numbers) - 1 
        while left < right:
            current_sum =  numbers[left] + numbers[right]
            if current_sum == target:
                 return [left + 1, right + 1]
            elif current_sum < target:
                left +=1
            else :
                right -=1 
sol = Solution()
print(sol.twoSum([2,7,11,15], 9))
print(sol.twoSum([2,3,4], 6))
print(sol.twoSum([-1,0], -1))


   

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int a = 0;
        int b = numbers.size() - 1;

        vector<int> res;
        while (a < b) {
            int sum = numbers[a] + numbers[b]; 
            if (sum < target) {
                a++;
            } else if (sum > target) {
                b--;
            } else {
                res.push_back(a + 1);
                res.push_back(b + 1);
                break;
            }
        }
        return res;
    }
};
